src/lcm/discrete_problem.py

```diff
@@ -0,0 +1,280 @@
+    elif shock_type == "extreme_value":
+        raise NotImplementedError("Extreme value shocks are not yet implemented.")
+    else:
+        raise ValueError(f"Invalid shock_type: {shock_type}.")
+
+    return partial(
+        func,
+        choice_axes=choice_axes,
+        choice_segments=choice_segments,
+        params=params,
+    )
+
+
+# ======================================================================================
+# Discrete problem with no shocks
+# ======================================================================================
+
+
+def _solve_discrete_problem_no_shocks(
+    cc_values: Array,
+    choice_axes: tuple[int, ...] | None,
+    choice_segments: SegmentInfo | None,
+    params: dict[str, Any],  # noqa: ARG001
+) -> Array:
+    """Aggregate conditional continuation values over discrete choices.
+
+    Args:
+        cc_values (jax.Array): Array with conditional continuation values. Has one
+            dimensions per state and discrete choice variable.
+        choice_axes (tuple[int, ...]): A tuple of indices representing the axes in the
+            value function that correspond to discrete choices. Returns None if there
+            are no discrete choice axes.
+        choice_segments (SegmentInfo): Dictionary with the entries "segment_ids" and
+            "num_segments". segment_ids are a 1d integer array that partitions the first
+            dimension of values into choice sets over which we need to aggregate.
+            "num_segments" is the number of choice sets. The segment_ids are assumed to
+            be sorted.
+        params (dict): Dictionary with model parameters.
+
+    Returns:
+        jax.Array: Array with aggregated continuation values. Has less dimensions than
+            cc_values if choice_axes is not None and is shorter in the first dimension
+            if choice_segments is not None.
+
+    """
+    out = cc_values
+    if choice_axes is not None:
+        out = out.max(axis=choice_axes)
+    if choice_segments is not None:
+        out = _segment_max_over_first_axis(out, choice_segments)
+
+    return out
+
+
+def _segment_max_over_first_axis(
+    data: Array,
+    segment_info: SegmentInfo,
+) -> Array:
+    """Calculate a segment_max over the first axis of data.
+
+    Wrapper around ``jax.ops.segment_max``.
+
+    Args:
+        data (jax.Array): Multidimensional jax array.
+        segment_info (SegmentInfo): Dictionary with the entries "segment_ids" and
+            "num_segments". segment_ids are a 1d integer array that partitions the
+            first dimension of `data` into segments over which we need to aggregate.
+            "num_segments" is the number of segments. The segment_ids are assumed to be
+            sorted.
+
+    Returns:
+        jax.Array: An array with shape (num_segments,) + data.shape[1:] representing the
+            segment maximums.
+
+    """
+    return jax.ops.segment_max(
+        data=data,
+        indices_are_sorted=True,
```

Comment:
Wouldn't it make sense to make this part of `segment_info`? You see it that it is always True at the point where it is defined then.

This function would then be gone, I guess.

Reply:
That is a good idea. I cannot add it to `SegmentInfo` right now because doing so would require me to change a large chunk of the code base that I want to refactor at some point. But I have noted it, and I will already delete the wrapper function.